multi-agent/lib/mas/elements/nodes/claude_agent/validator.py

```
@@ -0,0 +1,154 @@
+    - Model string is valid
+    - Permission mode is recognized
+    - Max turns bounds
+    - Retriever dependency (if configured)
+
+    Note: Vertex AI credential reachability is validated in real-time
+    via the claude_agent.validate_vertex_connection action (ActionHint).
+    """
+
+    VALID_PERMISSION_MODES = {
+        "default", "acceptEdits", "plan", "dontAsk", "bypassPermissions",
+    }
```

Ruff (0.15.15)

[warning] Mutable default value for class attribute

(RUF012)